src/Command/Pull/PullCommandBase.php

```diff
@@ -103,34 +103,43 @@ protected function pullCode(InputInterface $input, OutputInterface $output): voi
   /**
    * @param \Symfony\Component\Console\Input\InputInterface $input
    * @param \Symfony\Component\Console\Output\OutputInterface $output
+   * @param bool $on_demand Force on-demand backup.
+   * @param bool $download Skip import.
    *
    * @throws \Acquia\Cli\Exception\AcquiaCliException
-   * @throws \Exception
    */
-  protected function pullDatabase(InputInterface $input, OutputInterface $output): void {
-    $this->connectToLocalDatabase($this->getLocalDbHost(), $this->getLocalDbUser(), $this->getLocalDbName(), $this->getLocalDbPassword(), $this->getOutputCallback($output, $this->checklist));
+  protected function pullDatabase(InputInterface $input, OutputInterface $output, bool $on_demand, bool $download): void {
+    if (!$download) {
+      // Verify database connection.
+      $this->connectToLocalDatabase($this->getLocalDbHost(), $this->getLocalDbUser(), $this->getLocalDbName(), $this->getLocalDbPassword(), $this->getOutputCallback($output, $this->checklist));
```

Comment:
I vaguely remember this being problematic to check before.

Reply:
I agree it's problematic, but it's no different than what we do currently, I just stuck it in a conditional.